server/src/test/java/org/elasticsearch/common/joda/JavaJodaTimeDuellingTests.java

```diff
@@ -343,6 +343,17 @@ public void testDuellingFormatsValidParsing() {
         assertSameDate("2012-W1-1", "weekyear_week_day");
     }
 
+    public void testCompositeParsing(){
+        //in all these examples the second pattern will be used
+        assertSameDate("2014-06-06T12:01:02.123", "yyyy-MM-dd'T'HH:mm:ss||yyyy-MM-dd'T'HH:mm:ss.SSS");
```

Comment:
I think we need some more tests:
* 3 patterns
* random ordering of many patterns

Reply:
Happy to add more test cases. Do you have anything specific in mind? 
It won't harm adding testcases with 3 patterns, but I am not convinced we need a random ordering here. What scenario would it cover?